File: ios/StageViewManager.m
//
//  StageViewManager.m
//  ApplicationAmazonIVS
//
//  Created by macmini on 03/01/25.
//

#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(StageViewManager, RCTViewManager)

RCT_EXTERN_METHOD(flipCamera:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(toggleAudio:(nonnull NSNumber *)node isMuted:(BOOL)isMuted)

@end

File: ios/StageViewManager.swift
import Foundation
import React

@objc(StageViewManager)
class StageViewManager: RCTViewManager {
    override static func moduleName() -> String! {
        return "StageViewManager"
    }
    
    override class func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func view() -> UIView! {
        return StageView()
    }
    
    
    @objc func flipCamera(_ node: NSNumber) {
        DispatchQueue.main.async {
            if let view = self.bridge?.uiManager.view(forReactTag: node) as? StageView {
                view.flipCamera()
            }
        }
    }
    
    @objc func toggleAudio(_ node: NSNumber, isMuted: Bool) {
        DispatchQueue.main.async {
            if let view = self.bridge?.uiManager.view(forReactTag: node) as? StageView {
                view.toggleAudio(isMuted)
            }
        }
    }
}
